### PR TITLE
ISSUE 438 Email notification for sharee when delegating calendar

### LINKED ISSUE #438: Email notification for sharee when delegating calendar
WHEN Bob shares his calendar with Cedric
THEN Cedric get's an email notification `Bob shared his XXX calendar with you`

Not applicable to subscription created by Cedric himself.

CF https://ci.linagora.com/linagora/lgs/openpaas/linagora.esn.calendar/-/issues/1809

We can likely listen `calendar:calendar:updated` RabbitMQ exchange, filter out self created subscriptions.

Message payload is currently insuficient in order to implement this well, we would need to enrich the message:

```
{"calendarPath":"\/calendars\/67e26ebbecd9f300255a9f80\/dafa1cb9-17c6-40fa-bd0c-4da06f2faf17","calendarProps":{"access":"dav:read-write"}}
{"calendarPath":"\/calendars\/6053022c9da5ef001f430b43\/c432a882-2972-4542-8982-62052addad00","calendarProps":{"access":"dav:administration"}}
{"calendarPath":"\/calendars\/6053022c9da5ef001f430b43\/c432a882-2972-4542-8982-62052addad00","calendarProps":null}
```

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/twake-calendar-side-service/issues/438

## Summary by CodeRabbit

* **New Features**
  * Email notifications when a calendar is delegated: localized subject/body, logo, delegator & calendar details, and link.
  * New background consumer and reconnection handler to process delegated-calendar events with automatic restart on reconnect.
  * Queue registration for delegated-created events.

* **Localization**
  * Added English, French, Russian, and Vietnamese translations for delegation emails.

* **Tests**
  * New unit and integration tests for payload deserialization, delivery, localization, invalid payloads, and no-mail edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: calendar-amqp/src/main/java/com/linagora/calendar/amqp/DelegatedCalendarNotificationEnricher.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import com.linagora.calendar.amqp.CalendarDelegatedNotificationConsumer.CalendarDelegatedCreatedMessage;
+import com.linagora.calendar.dav.CalDavClient;
+import com.linagora.calendar.storage.CalendarURL;
+import com.linagora.calendar.storage.OpenPaaSUser;
+import com.linagora.calendar.storage.OpenPaaSUserDAO;
+import com.linagora.calendar.storage.configuration.resolver.SettingsBasedResolver;
+
+import reactor.core.publisher.Mono;
+
+public class DelegatedCalendarNotificationEnricher {
```
Comment (chibenwa) [inline calendar-amqp/src/main/java/com/linagora/calendar/amqp/DelegatedCalendarNotificationEnricher.java]: This naming sucks and  when naming sucks generally the abstraction beneaf it sucks too

How about `DelegatedCalendarNotificationData.Factory` ?


```
record DelegatedCalendarNotificationData(...) {
    static class Factory {
         Mono<DelegatedCalendarNotificationData> from(CalendarDelegatedCreatedMessage message) {
             ...
         } 
    }
}
```

?

Comment (vttranlina) [reply calendar-amqp/src/main/java/com/linagora/calendar/amqp/DelegatedCalendarNotificationEnricher.java]: I refactored it, 
I will merge logic build notification data inside the `DelegatedCalendarNotificationHandler` class.

Comment (vttranlina) [reply calendar-amqp/src/main/java/com/linagora/calendar/amqp/DelegatedCalendarNotificationEnricher.java]: I will push it in the next commit

**Review by chibenwa — APPROVED** [review]

**Review by hungphan227 — APPROVED** [review]
